### PR TITLE
Hybridize

### PR DESCRIPTION
**Pull request by dlockhart**

### CODE REVIEW

File: d2l-image-action-group.html [hunk 1]
```diff
@@ -1,30 +1,28 @@
...
-			:host-context([dir="rtl"]) ::content > a[is="d2l-image-action-link"] {
-				margin-right: 0;
-				margin-left: 1.2rem;
-			}
-			:host-context([dir="rtl"]) ::content > button[is="d2l-image-action"]:last-child,
-			:host-context([dir="rtl"]) ::content > a[is="d2l-image-action-link"]:last-child {
-				margin-left: 0;
+			.d2l-image-action-group-container ::slotted(d2l-image-action-button),
+			.d2l-image-action-group-container ::slotted(d2l-image-action-link) {
+				margin-right: 1.2rem;
```
Comment (dlockhart) [inline d2l-image-action-group.html]: Had lots of issues with how we approach RTL and `::slotted`. I could get it working in Polymer 2 shadow, and Polymer 1 shadow/shady, but not Polymer 2 shady.

I ended up just changing the approach here entirely -- instead of flipping the styles in RTL, I just kill the right-margin on the last item with a negative margin on the outer container. No need for flipping.

File: d2l-image-action-link.html [hunk 2]
```diff
@@ -1,42 +1,74 @@
...
 				@apply --d2l-image-action-icon;
 			}
+			/* Polymer 2: shady DOM */
+			.d2l-image-action-icon:dir(rtl) {
+				@apply --d2l-image-action-icon-rtl;
+			}
+			/* Polymer 1: shady/shadow DOM, Polymer 2: shadow DOM */
 			:host-context([dir="rtl"]) .d2l-image-action-icon {
 				@apply --d2l-image-action-icon-rtl;
 			}
```
Comment (dlockhart) [inline d2l-image-action-link.html]: Good times in RTL-land. This combination was the only one that worked consistently across the 4 permutations of Polymer 1.x/2.0 and shadow/shady.

File: d2l-image-action-shared-styles.html [hunk 3]
```diff
@@ -10,7 +10,6 @@
 					border: none;
 					cursor: pointer;
 					display: inline-block;
-					font-family: inherit; /* issue with native CSS props: https://github.com/Polymer/polymer/issues/4282 */
```
Comment (dlockhart) [inline d2l-image-action-shared-styles.html]: I moved this from the shared styles (applied to button and link) to just be applied to button.

File: d2l-image-action.html [hunk 4]
```diff
@@ -1,50 +1,3 @@
-<link rel="import" href="../polymer/polymer.html">
-<link rel="import" href="../d2l-icons/d2l-icon.html">
-<link rel="import" href="d2l-image-action-shared-styles.html">
+<link rel="import" href="d2l-image-action-button.html">
```
Comment (dlockhart) [inline d2l-image-action.html]: This file just becomes one you can include to import all 3.

File: bower.json [hunk 5]
```diff
@@ -12,17 +12,48 @@
...
+    "d2l-link": "^4.0.0",
+    "d2l-typography": "^6.0.0 || ^5.4.0",
+    "polymer": "Polymer/polymer#1.9 - 2"
   },
   "devDependencies": {
-    "d2l-demo-template": "^0.0.9"
+    "iron-component-page": "PolymerElements/iron-component-page#1 - 2",
+    "iron-demo-helpers": "PolymerElements/iron-demo-helpers#1 - 2",
+    "web-component-tester": "^6.0.0",
+    "webcomponentsjs": "webcomponents/webcomponentsjs#^1.0.0",
```
Comment (awikkerink) [inline bower.json]: Why is this here and in the resolutions?

Comment (dlockhart) [inline bower.json]: I guess `devDependencies` don't need resolutions?

Comment (awikkerink) [inline bower.json]: Resolutions work for `devDependencies` as well. I'm not sure I understand

The webcomponents resolution is needed because bower gets confused. But you shouldn't need to specify it in `devDependencies`

File: d2l-image-action-group.html (hunk 1 above)
Comment (dbatiste) [inline d2l-image-action-group.html]: Looks like a similar sort of scenario with [floating workflow buttons](https://github.com/BrightspaceUI/button/blob/master/d2l-floating-buttons.html#L40).  I wonder if it has the same issues you encountered?

Comment (dlockhart) [inline d2l-image-action-group.html]: It's possible.
